### PR TITLE
Fix various typos [skip ci]

### PR DESCRIPTION
**Pull request by luzpaz**
Fixed via 
```
codespell -q 3 -L aci,ake,aline,alle,alledges,alocation,als,ang,anid,ba,beginn,behaviour,bloaded,byteorder,calculater,cancelled,cancelling,cas,cascade,centimetre,childs,colour,colours,commen,currenty,dof,doubleclick,dum,eiter,elemente,feld,freez,hist,iff,indicies,initialisation,initialise,initialised,initialises,initialisiert,ist,kilometre,lod,mantatory,methode,metres,millimetre,modell,nd,noe,normale,normaly,nto,numer,oder,orgin,orginx,orginy,ot,pard,pres,programm,que,recurrance,rougly,seperator,serie,sinc,strack,substraction,te,thist,thru,tread,uint,unter,vertexes,wallthickness,whitespaces -S ./.git,*.po,*.ts,./ChangeLog.txt,./src/3rdParty,./src/Mod/Assembly/App/opendcm,./src/CXX,./src/zipios++,./src/Base/swig*,./src/Mod/Robot/App/kdl_cp,./src/Mod/Import/App/SCL,./src/WindowsInstaller,./src/Doc/FreeCAD.uml
```

### CODE REVIEW

File: src/Mod/Assembly/Gui/TaskAssemblyConstraints.ui [hunk 1]
```diff
@@ -192,7 +192,7 @@
          </sizepolicy>
         </property>
         <property name="toolTip">
-         <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Fixes the first geometry in its rotation and translation. Note that fix only works its the direct parent assembly. If you stack assemblys, the parent assembly will not be fixed inside the other ones.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
+         <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Fixes the first geometry in its rotation and translation. Note: this fix only works its direct parent assembly. If you stack assemblies, the parent assembly will not be fixed inside the other ones.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
```
Comment (luzpaz) [inline src/Mod/Assembly/Gui/TaskAssemblyConstraints.ui]: please review

Comment (vocx-fc) [reply src/Mod/Assembly/Gui/TaskAssemblyConstraints.ui]: I would suggest not to touch this part of the code, as it is inside the Assembly workbench, which is unused at the moment. Just leave it as is.

File: src/Mod/MeshPart/App/AppMeshPartPy.cpp [hunk 2]
```diff
@@ -134,8 +134,8 @@ class Module : public Py::ExtensionModule<Module>
             "    MinLength (required, float)\n"
             "    Fineness (required, integer)\n"
             "    SecondOrder (optional, integral boolean)\n"
-            "    Optimize (optional, integeral boolean)\n"
-            "    AllowQuad (optional, integeral boolean)\n"
+            "    Optimize (optional, integral boolean)\n"
+            "    AllowQuad (optional, integral boolean)\n"
```
Comment (vocx-fc) [inline src/Mod/MeshPart/App/AppMeshPartPy.cpp]: This is maybe "integer boolean", talking about a non-zero natural number, like 1, 2, 3, 4, 5, etc., to indicate `True`, and the zero indicates `False`.

Comment (luzpaz) [reply src/Mod/MeshPart/App/AppMeshPartPy.cpp]: @wmayer care to weigh in?

Comment (vocx-fc) [reply src/Mod/MeshPart/App/AppMeshPartPy.cpp]: Yes, this is definitely an integer boolean.

In that same file, starting in line 476 you can see the implementation of the function `meshFromShape`. The specific code that uses the arguments `SecondOrder`, `Optimize`, and `AllowQuad` starts in line 579.
```cpp
#if defined (HAVE_NETGEN)
        static char* kwds_fineness[] = {"Shape", "Fineness", "SecondOrder", "Optimize", "AllowQuad",NULL};
        PyErr_Clear();
        int fineness=0, secondOrder=0, optimize=1, allowquad=0;
        if (PyArg_ParseTupleAndKeywords(args.ptr(), kwds.ptr(), "O!i|iii", kwds_fineness,
                                        &(Part::TopoShapePy::Type), &shape, &fineness,
                                        &secondOrder, &optimize, &allowquad)) {
            MeshPart::Mesher mesher(static_cast<Part::TopoShapePy*>(shape)->getTopoShapePtr()->getShape());
            mesher.setMethod(MeshPart::Mesher::Netgen);
            mesher.setFineness(fineness);
            mesher.setSecondOrder(secondOrder != 0);
            mesher.setOptimize(optimize != 0);
            mesher.setQuadAllowed(allowquad != 0);
            return Py::asObject(new Mesh::MeshPy(mesher.createMesh()));
        }
```
As you can see, they are defined as `int`. Moreover, the arguments are passed to the `mesher` methods using a test, `mesher.setSecondOrder(secondOrder != 0)`.

What this does is convert the integer value into a boolean value. If `secondOrder` is in fact `0`, the test will return `False`, which is what will be passed to the function. If `secondOrder` is any other integer, the test will return `True`, which will be passed to the function.

If we go into the `Mesher.h` source, we see that the methods `setSecondOrder` et al., are defined to accept a boolean.
```cpp
    void setSecondOrder(bool on)
    { secondOrder = on; }
    void setOptimize(bool on)
    { optimize = on;}
    void setQuadAllowed(bool on)
    { allowquad = on;}
```

Comment (luzpaz) [reply src/Mod/MeshPart/App/AppMeshPartPy.cpp]: Nice breakdown! I added the revisions to 20a27a9

**Comment by luzpaz** [comment]
These should of been squashed before merging :man_facepalming:

**Pull request merged**